lib/react/rails/view_helper.rb

```diff
@@ -1,28 +1,28 @@
 module React
   module Rails
     module ViewHelper
-
       # Render a UJS-type HTML tag annotated with data attributes, which
       # are used by react_ujs to actually instantiate the React component
       # on the client.
-      #
-      def react_component(name, args = {}, options = {}, &block)
+      def react_component(name, props = {}, options = {}, &block)
         options = {:tag => options} if options.is_a?(Symbol)
-        block = Proc.new{concat React::Renderer.render(name, args)} if options[:prerender]
+
+        if options[:prerender]
+          block = Proc.new{ concat React::ServerRendering.render(name, props) }
+        end
 
         html_options = options.reverse_merge(:data => {})
         html_options[:data].tap do |data|
           data[:react_class] = name
-          data[:react_props] = React::Renderer.react_props(args) unless args.empty?
+          data[:react_props] = (props.is_a?(String) ? props : props.to_json)
```

Comment:
This should always be JSON because `ReactRailsUJS` will try to parse it as JSON!